src/main/java/hudson/plugins/gradle/injection/CopyUtil.java

```diff
@@ -11,11 +12,28 @@ private CopyUtil() {
     }
 
     public static void copyResourceToNode(FilePath nodePath, String resourceName) throws IOException, InterruptedException {
-        try (InputStream libIs = CopyUtil.class.getResourceAsStream("/hudson/plugins/gradle/injection/" + resourceName)) {
-            if (libIs == null) {
+        doWithResource(resourceName, is -> {
+            nodePath.copyFrom(is);
+            return null;
+        });
+    }
+
+    public static String resourceDigest(String resourceName) throws IOException, InterruptedException {
+        return doWithResource(resourceName, Util::getDigestOf);
+    }
+
+    private static <T> T doWithResource(String resourceName, RemoteFunction<InputStream, T> action) throws IOException, InterruptedException {
+        try (InputStream is = CopyUtil.class.getResourceAsStream("/hudson/plugins/gradle/injection/" + resourceName)) {
+            if (is == null) {
                 throw new IllegalStateException("Could not find resource: " + resourceName);
             }
-            nodePath.copyFrom(libIs);
+            return action.apply(is);
         }
     }
+
+    @FunctionalInterface
+    private interface RemoteFunction<T, R> {
```

Comment:
I think remote is a bit misleading, maybe `CheckedFunction` is a better name as its only purpose is that it can throw exceptions.

Reply:
Or just `ThrowingFunction`

Reply:
OK, let's rename it.

Reply:
Renamed.